charts/telemetry-collector/templates/configmap.yaml

```diff
@@ -0,0 +1,58 @@
+{{- include
+      "accelleran.common.configMap"
+      (fromYaml (include "accelleran.telemetry-collector.configMap.args" .))
+-}}
+
+{{- define "accelleran.telemetry-collector.configMap.args" -}}
+{{- $ := . -}}
+
+top:
+  {{ $ | toYaml | nindent 2 }}
+
+name: {{ include "accelleran.common.fullname" (dict "top" $) }}-config
+
+data:
+  config.yaml: |-
+    internal:
+      kafka:
+        url: "{{ include "accelleran.common.bootstrap.kafka.hostname" (dict "top" .) }}:{{ include "accelleran.common.bootstrap.kafka.port" (dict "top" .) }}"
+      {{- if $.Values.config.influxDB }}
+      influxDB:
+        url: {{ tpl $.Values.config.influxDB.url $ | required "The influxdb URL needs to be provided (config.influxDB.url)" | quote }}
+        token: ""
+        org: "{{ $.Values.config.influxDB.org }}"
+      {{- end }}
+    pipelines:
+      {{- $natsHostname := include "accelleran.common.bootstrap.nats.hostname" (dict "top" .) }}
+      {{- $natsPort := include "accelleran.common.bootstrap.nats.port" (dict "top" .) }}
+      {{- range $pipeline := $.Values.config.pipelines }}
+      - name: "{{ $pipeline.name }}" 
+        component: "{{ $pipeline.component }}"
+        vendor: "{{ $pipeline.vendor }}"
+        enabled: "{{ $pipeline.enabled }}"
+        {{- if $pipeline.consumers }}
+        consumers:
+          {{- if $pipeline.consumers.natsConsumer }}
+            natsConsumer:
+              url: {{ $pipeline.consumers.natsConsumer.url | default (printf "%s:%v" $natsHostname $natsPort) | quote }}
+              subjects: "{{ $pipeline.consumers.natsConsumer.subjects }}"
+          {{- end }}
+          {{- with $pipeline.consumers.udpConsumer }}
+            udpConsumer: {{ . | toYaml | nindent 14 }}
+          {{- end }}
+        {{- end }}
+        {{- with $pipeline.publishers }}
+          {{ if and $pipeline.publishers.kafkaPublisher (not $pipeline.publishers.kafkaPublisher.topic) }}
+            {{ fail "When specifying a kafka publisher also a kafka topic needs to be provided" }}
+          {{- end }}
+          {{ if and $pipeline.publishers.influxPublisher (not $pipeline.publishers.influxPublisher.buckets) }}
+            {{ fail "When specifying a influx publisher also a buckets needs to be provided" }}
+          {{- end }}
+        publishers: {{ . | toYaml | nindent 10 }}
+        {{- end }}
+        {{- with $pipeline.vendorConfig }}
+          vendorConfig: {{ . | toYaml | nindent 10 }}
+        {{- end }}
+      {{- end }}
```

Comment:
I guess we should basically have this whole structure in the values file so we can really easily adapt to new config etc (basically like the comments I gave on the udpConsumer and publishers). Not sure if it will actually work or that we need to have a string with all the templating inside instead of actual yaml (due to the conditions around some blocks, lists, ...).

```suggestion
  config.yaml: |-
    {{- if and $.Values.config.internal.influxDB (tpl $.Values.config.internal.influxDB.url $) -}}
      {{- fail "The influxdb URL needs to be provided (config.internal.influxDB.url)" -}}
    {{- end -}}
    {{- range $index, $pipeline := $.Values.config.pipelines -}}
      {{- if and $pipeline.publishers $pipeline.publishers.kafkaPublisher (not $pipeline.publishers.kafkaPublisher.topic) -}}
        {{ fail (printf "When specifying a kafka publisher also a kafka topic needs to be provided (config.pipelines[%d].publishers.kafkaPublisher.topic" $index) }}
      {{- end -}}
      {{- if and $pipeline.publishers $pipeline.publishers.influxPublisher (not $pipeline.publishers.influxPublisher.buckets) -}}
        {{ fail (printf "When specifying an influx publisher also buckets need to be provided (config.pipelines[%d].publishers.influxPublisher.buckets" $index) }}
      {{- end -}
    {{- end -}}
    {{ tpl $.Values.config $ | toYaml | nindent 4 }}
```

In `values.yaml` then:

```yaml
config:
  internal:
    kafka:
      url: "{{ include "accelleran.common.bootstrap.kafka.hostname" (dict "top" .) }}:{{ include "accelleran.common.bootstrap.kafka.port" (dict "top" .) }}"
    influxDB:
      url: ""
      token: ""
      org: ""
  pipelines:
    - name: "cu_measurements"
      component: "cu"
      vendor: "accelleran"
      enabled: "true"
      consumers:
        natsConsumer:
          url: {{ $pipeline.consumers.natsConsumer.url | default (printf "%s:%v" (include "accelleran.common.bootstrap.nats.hostname" (dict "top" .)) (include "accelleran.common.bootstrap.nats.port" (dict "top" .))) | quote }}
          subjects: "*.5G_MEAS_INFO.*.*.*.*"
      publishers:
        kafkaPublisher:
          topic: "accelleran.drax.5g.ric.o1.ves"
      influxPublisher:
        defaultBucket: "meas5g"
        buckets:
          RrcMeasurementReport: "rrc_measurements"
    - name: "pm_counters"
      component: "cu"
      vendor: "accelleran"
      enabled: "true"
      consumers:
        natsConsumer:
          url: {{ $pipeline.consumers.natsConsumer.url | default (printf "%s:%v" (include "accelleran.common.bootstrap.nats.hostname" (dict "top" .)) (include "accelleran.common.bootstrap.nats.port" (dict "top" .))) | quote }}
          subjects: "*.CUCP_COUNTERS_INFO,*.CUUP_COUNTERS_INFO"
      publishers:
        kafkaPublisher:
          topic: "accelleran.drax.5g.ric.o1.ves"
        influxPublisher:
          defaultBucket: "counters5g"
          buckets:
            RRCConnEstabAtt: "rrc_connection_metrics"
            RRCConnEstabSucc: "rrc_connection_metrics"
            MMHoExeIntraReq: "handover_and_mobility_metrics"
            MMHoExeIntraSucc: "handover_and_mobility_metrics"
            RRCConnMean: "rrc_connection_metrics"
            RRCConnMax: "rrc_connection_metrics"
            SMPDUSessionSetupReq: "pdu_session_management_metrics"
            SMPDUSessionSetupReqSnssai: "pdu_session_management_metrics"
            SMPDUSessionSetupFail: "pdu_session_management_metrics"
            L1MSSRSRP: "rrc_measurements"
            MMHoPrepInterReq: "handover_and_mobility_metrics"
            MMHoPrepInterSucc: "handover_and_mobility_metrics"
            MMHoPrepInterFail: "handover_and_mobility_metrics"
            MMHoResAlloInterReq: "handover_and_mobility_metrics"
            MMHoResAlloInterSucc: "handover_and_mobility_metrics"
            MMHoResAlloInterFail: "handover_and_mobility_metrics"
            MMHoExeInterReq: "handover_and_mobility_metrics"
            AccDRBEstabAtt5QiSnssai: "radio_resource_utilization_metrics"
            DRBEstabSuccSnssai: "radio_resource_utilization_metrics"
            DRBEstabSucc5Qi: "radio_resource_utilization_metrics"
            DRBRelActNbr5QiSnssai: "radio_resource_utilization_metrics"
            DRBRelActNbrSnssai: "radio_resource_utilization_metrics"
            DRBRelActNbr5Qi: "radio_resource_utilization_metrics"
            MMHoExeInterReqTimeMean: "handover_and_mobility_metrics"
            MMHoExeInterReqTimeMax: "handover_and_mobility_metrics"
            AccGTPThpDl: "throughput_metrics"
            AccGTPThpDlQfiSnssaiPlmn: "throughput_metrics"
            AccGTPThpUl: "throughput_metrics"
            AccGTPThpUlQfiSnssaiPlmn: "throughput_metrics"
            AccDRBPdcpSduDelayDlQfiSnssaiPlmn: "radio_resource_utilization_metrics"
            AccDRBF1UPacketLossRateUlQoS: "radio_resource_utilization_metrics"
            AccDRBF1UPacketLossRateUlQfiSnssaiPlmn: "radio_resource_utilization_metrics"
            AccDRBF1UPacketLossRateUlSnssai: "radio_resource_utilization_metrics"
            AccDRBPdcpPacketDropRateDlQoS: "radio_resource_utilization_metrics"
            AccDRBPdcpPacketDropRateDlSnssai: "radio_resource_utilization_metrics"
            AccDRBPdcpPacketDropRateDlQfiSnssaiPlmn: "radio_resource_utilization_metrics"
            DRBPdcpSduDelayDlQoS: "radio_resource_utilization_metrics"
            DRBPdcpSduDelayDlSnssai: "radio_resource_utilization_metrics"
            DRBPdcpSduDelayDlDistQos: "radio_resource_utilization_metrics"
            DRBPdcpSduDelayDlDist: "radio_resource_utilization_metrics"
            DRBPdcpSduDelayDlDistSnssai: "radio_resource_utilization_metrics"
            AccDRBPdcpSduDelayDlDistQfiSnssaiPlmn: "radio_resource_utilization_metrics"
    - name: "acc_du"
      component: "du"
      vendor: "accelleran"
      enabled: "true"
      consumers:
        udpConsumer:
          port: "55555"
      publishers:
        kafkaPublisher:
          topic: "accelleran.drax.5g.ric.o1.ves"
        influxPublisher:
          defaultBucket: "acc_du"
          buckets:
            DUUEMeasurements: "du_ue_measurements"
            DURLCMeasurements: "du_rlc_measurements"
```

Reply:
It's done slightly differently then proposed as the kafka and nats url aren't injected with the templating. I left a separate comment for nats, but the initial point here is resolved.